### DISCUSSION TITLE
we shouldn't need scene inheritance

### DISCUSSION THREAD

**Discussion by BaddRadish** [godotengine/godot-proposals#5932] [Editor]
- we can instantiate a scene as a child node
- but we cant make it scene root
- but we /can/ create an "inherited scene" and then it /is/ scene root
- but then we cant reparent a regular node as the scene root

none of this is necessary. just treat scenes like nodes consistently. let them be root or be moved like any other scene. whats the problem?? why are we creating a pointless distinction here?

if there /is/ some technical reason for this then lets not make the user have to deal with it. just detect if the root node of a scene is another scene and handle it silently.

**Comment by Zireael07** [comment]
> just detect if the root node of a scene is another scene and handle it silently.

I don't understand this sentence... how can the root be another scene?

**Comment by BaddRadish** [comment]
if the root node is an instanced OR inherited scene

**Comment by theraot** [comment]
Adding the ability to convert convert between scene inheritance and composition to Godot makes sense. For example: If you tell the editor to make an scene instance the new scene root, Godot could ask you if you want to convert the scene to an inherited scene. Similarly, if you have an inherited scene and you tell Godot to make some other `Node` the root, Godot could ask you if you want to convert the scene inheritance into composition. *And yes, it might even be silent (without asking)*.

**Comment by timothyqiu** [comment]
Related: #1122

> if there /is/ some technical reason for this then lets not make the user have to deal with it. just detect if the root node of a scene is another scene and handle it silently.

I think it'll be handy if we can handle it silently too. But as far as I can tell, it's not as simple as it seems. Regular nodes can be made root or non-root at anytime as they are not related to each other. However, for an inherited scene, it is likely that some hierarchy nodes are provided by the root node.

```
- Bob (Inherits "Character.tscn" Scene)
  - Graphics (from "Character.tscn")
    - Sprite2D
  - Detector (Instanced from "Detector.tscn" whose root is an Area2D)
  - Node2D
    - Label
```

It's easy to make Detector the root if it's not an inherited scene:

```
- Detector
  - Graphics
    - Sprite2D
  - Bob
  - Node2D
    - Label
```

But since that "Graphics" node is inherited from "Character.tscn" (not exist in "Bob" scene), it's hard to decide where to put the Sprite2D node.

It's also possible for nodes in "Detector.tscn" to conflict with existing nodes. e.g. If it also has a node at `Node2D/Label`, then the existing `Node2D/Label` has to be renamed.
